modules/tensor_mechanics/src/postprocessors/SidesetForce.C

```diff
@@ -0,0 +1,46 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "SidesetForce.h"
+#include "metaphysicl/raw_type.h"
+
+registerMooseObject("TensorMechanicsApp", SidesetForce);
+registerMooseObject("TensorMechanicsApp", ADSidesetForce);
+
+template <bool is_ad>
+InputParameters
+SidesetForceTempl<is_ad>::validParams()
+{
+  InputParameters params = SideIntegralPostprocessor::validParams();
+  params.addClassDescription(
+      "This postprocessor computes the force on sidesets in parameter direction");
+  params.addRequiredParam<MaterialPropertyName>("stress_tensor", "The rank two stress tensor name");
+  params.addRequiredParam<RealVectorValue>("direction",
+                                           "Direction in which the force is to be computed");
+  params.set<bool>("use_displaced_mesh") = true;
+  return params;
+}
+
+template <bool is_ad>
+SidesetForceTempl<is_ad>::SidesetForceTempl(const InputParameters & parameters)
+  : SideIntegralPostprocessor(parameters),
+    _tensor(getGenericMaterialProperty<RankTwoTensor, is_ad>("stress_tensor")),
+    _dir(getParam<RealVectorValue>("direction"))
+{
+}
+
+template <bool is_ad>
+Real
+SidesetForceTempl<is_ad>::computeQpIntegral()
+{
+  return _normals[_qp] * (MetaPhysicL::raw_value(_tensor[_qp]) * _dir);
```

Comment:
I think you should use `_dir` instead of `_normals[_qp]` in this equation.

Reply:
But I use both.